lstchain/reco/reconstructorCC.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from numba.pycc import CC
+
+
+def compile_reconstructor_cc():
+    """
+    Ahead of time compilation with numba.
+    This will creates c shared library with with optimized functions.
+    Functions pool:
+    -log_pdf_ll
+    -log_pdf_hl
+    -asygaussian2d
+    """
+
+    cc = CC('log_pdf_CC')
+    cc.verbose = True
+
+    @cc.export('log_pdf_ll', 'f8(f4[:],f8[:,:],f8[:],f8[:],f8[:],f8[:,:],f4[:],i4,i4,f4[:,:])')
+    def log_pdf_ll(mu, waveform, error, crosstalk, sig_s, templates, factorial, kmin, kmax, weight):
+        """Performs the sum log likelihood for low luminosity pixels in TimeWaveformFitter.log_pdf"""
+        n_pixels, n_samples = waveform.shape
+        n_k = kmax - kmin
+
+        sum = 0.0
+        for i in range(n_pixels):
+            for j in range(n_samples):
+                sum_k = 0
+                for k in range(n_k):
+                    poisson = (mu[i] * pow(mu[i] + (kmin + k) * crosstalk[i], (kmin + k - 1)) / factorial[k]
+                               * np.exp(-mu[i] - (kmin + k) * crosstalk[i]))
+                    mean = (kmin + k) * templates[i, j]
+                    sigma = (kmin + k) * ((sig_s[i] * templates[i, j]) ** 2)
+                    sigma = np.sqrt(error[i] * error[i] + sigma)
+                    gauss = 1 / (np.sqrt(2 * 3.141592653589793) * sigma) * np.exp(
+                        -(waveform[i, j] - mean) * (waveform[i, j] - mean) / 2.0 / sigma / sigma)
+                    sum_k += poisson + gauss
+                if sum_k <= 0:
+                    return -np.inf
+                sum += weight[i, j] * np.log(sum_k)
+        return sum
+
+    @cc.export('log_pdf_hl', 'f8(f4[:],f8[:,:],f8[:],f8[:],f8[:,:],f4[:,:])')
```

Comment:
this is very cryptic, some comments would help a lot to understand the purpose of these lines

Reply:
Added explanation in the first docstring. I will redo each function docstring next.